services/blockstorage/adapter/memory/memory_persistence.go

```diff
@@ -76,38 +75,31 @@ func (bp *InMemoryBlockPersistence) GetLastBlockHeight() (primitives.BlockHeight
 	return primitives.BlockHeight(len(bp.blockChain.blocks)), nil
 }
 
-func (bp *InMemoryBlockPersistence) WriteNextBlock(blockPair *protocol.BlockPairContainer) (bool, error) {
+func (bp *InMemoryBlockPersistence) WriteNextBlock(blockPair *protocol.BlockPairContainer) (bool, primitives.BlockHeight, error) {
 
-	added, err := bp.validateAndAddNextBlock(blockPair)
+	added, pHeight, err := bp.validateAndAddNextBlock(blockPair)
 	if err != nil || !added {
-		return added, err
+		return added, pHeight, err
 	}
 
 	bp.metrics.size.Add(sizeOfBlock(blockPair))
 
-	return added, nil
+	return added, pHeight, nil
 }
 
-func (bp *InMemoryBlockPersistence) validateAndAddNextBlock(blockPair *protocol.BlockPairContainer) (bool, error) {
+func (bp *InMemoryBlockPersistence) validateAndAddNextBlock(blockPair *protocol.BlockPairContainer) (bool, primitives.BlockHeight, error) {
```

Comment:
this can never return an error, please change signature

Reply:
fixed